### PR TITLE
Adding perk form (to add a perk to the system)

### PR DESCRIPTION
**Pull request by nicholasyung1**
Adding new svelte page to act as a form for adding new perks and backend for adding new perks to the database.

### CODE REVIEW

File: client/src/components/AddPerk.svelte [hunk 1]
```diff
@@ -0,0 +1,98 @@
+<script>
+    import { onMount } from 'svelte';
+    import { goto } from '$app/navigation';
```
Comment (Triguy12) [inline client/src/components/AddPerk.svelte:3]: I don't think app/navigation will be needed anymore, routing can be set up through svelte pages similar to the Home.svelte page navigation.

Comment (Zhangjason308) [reply client/src/components/AddPerk.svelte:3]: Yeah this PR is blocked by my current PR that loads spring boot locally

File: server/src/main/java/PerkController.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+@RestController
+@RequestMapping("/api/addperks")
+@CrossOrigin(origins = "http://localhost:5173")
+public class PerkController {
+    @Autowired
+    private PerkRepository perkRepository;
+
+    @PostMapping
+    public void addNewPerk(@RequestBody Perk perk) {
```
Comment (Zhangjason308) [inline server/src/main/java/PerkController.java:12]: You may need to update your controller because I modified the Perk class

**Comment by Zhangjason308** [comment]
Merge main into your branch. Its currently outdated since we add alot of new changes

File: server/pom.xml
Comment (Waluiyeegi) [inline server/pom.xml:1]: Make sure to not push any of the XML files

File: client/src/components/AddPerk.svelte
Comment (Waluiyeegi) [inline client/src/components/AddPerk.svelte:1]: not urgent, but maybe try to make it a bit fancier or similar color schemes to our Home page

**Review by Zhangjason308 — APPROVED** [review]

File: client/src/components/AddPerk.svelte [hunk 3]
```diff
@@ -0,0 +1,103 @@
+<script>
+    import { onMount } from 'svelte';
+    import { Link } from "svelte-routing";
+
+
+        // Form data for the new perk
+        let perk = {
```
Comment (Zhangjason308) [inline client/src/components/AddPerk.svelte:7]: Can you check that this aligns with all the properties that Caleb has in his Perk.Svelte

File: client/src/components/AddPerk.svelte [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        expiryDate: '',
+    };
+
+        // Submit the form to the backend
+        async function addPerk() {
+        try {
+        const response = await fetch('http://localhost:8080/api/perks/addNewPerk', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(perk),
```
Comment (Zhangjason308) [inline client/src/components/AddPerk.svelte:21]: You may get an error if he did not account for the properties you added

Comment (Zhangjason308) [reply client/src/components/AddPerk.svelte:21]: We should follow up on a separate PR to get the form to work

**Review by Waluiyeegi — APPROVED** [review]
looks good

**Comment by Waluiyeegi** [comment]
looks 😊

**Comment by Zhangjason308** [comment]
Please fix merge conflicts to merge

**Pull request merged**